src/pins/gen.rs

```diff
@@ -52,12 +52,21 @@ macro_rules! pins {
             ///
             /// [`Pin`]: struct.Pin.html
             #[allow(non_camel_case_types)]
+            #[derive(Clone, Copy)]
```

Comment:
Nope! This means any pin type can be copied around without restrictions. Not sure what the actual consequence of that is, since the GPIO case at least still requires a token, but the API is certainly written with the assumption that pin types are unique. This is bound to break something, at the very least in the SWM API.

Reply:
Aaaah I was worried about that. The reason I added this was because I couldn't for the life of me get the `pin_info()` data out of `token` otherwise, but with your suggestion to just use `Pin`'s `ty`, which holds the same information, this should be resolved. Neat!

Reply:
I think having the pin information in the token was a stupid idea in the first place (I'm allowed to say that, I implemented it :smile:). It's confusing, and there never was any need for it. It just should have been a `PhantomData` from the start.

Glad that's going to get resolved with this PR!